apollo-ios/Sources/Apollo/ExecutionSources/NetworkResponseExecutionSource.swift

```diff
@@ -9,6 +9,8 @@ public struct NetworkResponseExecutionSource: GraphQLExecutionSource, CacheKeyCo
   public typealias RawObjectData = JSONObject
   public typealias FieldCollector = DefaultFieldSelectionCollector
 
+  public var shouldAttemptDeferredFragmentExecution: Bool { false }
```

Comment:
I wonder if we should work this property into the network response parsing of incremental responses? Looking at this plainly you'd assume that network responses should not execute deferred fragments but they do because currently that's decoupled from this flag.

Reply:
I'm confused. How are network responses executing deferred fragments?

Reply:
It's a few things:
1. If the operation is not a `subscription` then the `multipart/mixed` `accept` header is always added which lets the server respond with multipart responses if it can, such as with `@defer` operations - [here](https://github.com/apollographql/apollo-ios-dev/blob/feature/defer-execution-networking/apollo-ios/Sources/Apollo/RequestChainNetworkTransport.swift#L110).
2. The `IncrementalJSONResponseParsingInterceptor` is used in the request chain if the operation contains deferred fragments - [here](https://github.com/apollographql/apollo-ios-dev/blob/feature/defer-execution-networking/apollo-ios/Sources/Apollo/DefaultInterceptorProvider.swift#L48).
3. Then the partial vs. incremental JSON parsing is determined by whether we have parsed an initial (partial) response yet.

Reply:
So considering those steps above, they are really for multipart parsing which _should_ be decoupled from defer specifically. The thing that determines `@defer` parsing is the specification we send in the `accept` header and sent back to us in the response which we then use to determine which parser to use [here](https://github.com/apollographql/apollo-ios-dev/blob/feature/defer-execution-networking/apollo-ios/Sources/Apollo/MultipartResponseParsingInterceptor.swift#L67).

It seems odd to have a 'master' flag `shouldAttemptDeferredFragmentExecution` for defer networking parsing; I'm not sure where we'd put the check?

Reply:
That makes me think that the [check for defer operations](https://github.com/apollographql/apollo-ios-dev/blob/feature/defer-execution-networking/apollo-ios/Sources/Apollo/DefaultInterceptorProvider.swift#L48) to decide whether to use `IncrementalJSONResponseParsingInterceptor` vs. `JSONResponseParsingInterceptor` is also incorrect since subscriptions need incremental parsing too but won't be considered a 'defer' operation.

Reply:
OK - I'm going to keep those out of this PR and follow up with a defer-multipart-cleanup PR where I'll tackle these issues.

Reply:
For anyone else reading this - we've cleared this up in the documentation of the new property.